### ISSUE TITLE
Feature Request: Adding toJson methods to DTOs

### ISSUE THREAD

**Issue by hasimyerlikaya** [qonversion/flutter-sdk#372]
Hi,

I saw that `toJson` methods are disabled for DTOs. 
I don't know why you decided this, but it would be better if you added toJson methods too.

I was debugging with breakpoints, and at one point I decided to print the entitlement response to the console to do better and faster debugging. I realize that there is no toJson method for [entitlement.dart](https://github.com/qonversion/flutter-sdk/blob/develop/lib/src/dto/entitlement.dart). 

In some cases we might need to store the conversion responses to our databases. ToJson methods would be helpful for that too.

Can I add toJson methods to DTOs?

**Comment by SpertsyanKM** [comment]
Hi @hasimyerlikaya,

We were looking into implementing `toJson` and `fromJson` methods for internal use, aiming to facilitate data transfer to native code and to parse responses from it. I appreciate your interest in these methods as well. I’ll bring this up with the team, and we’ll look to incorporate this enhancement if there are no obstacles.

**Comment by hasimyerlikaya** [comment]
In fact, you already support fromJson methods. 
For some reason, the functionality for creating the toJson method has been disabled.

Thank you for your interest.

<img width="696" alt="image" src="https://github.com/user-attachments/assets/d080ede0-aa3e-4396-b44b-7ef98e1b80f9">

**Comment by SpertsyanKM** [comment]
Yes, we’ve implemented this for all entities that can be parsed from native bridges, but not for everything. The toJson methods have only been added for those that are transferred to native bridges.

If we decide to make both methods publicly available, we’ll include them for all entities. However, I foresee some challenges with this approach regarding changes. Currently, we can modify the keys and content as needed. Making these methods public would mean that any such changes would have to be reserved for major releases.

Anyway, I’ll talk it over with my colleagues first.

**Comment by hasimyerlikaya** [comment]
Honestly, I don’t think making changes to all models would be a well-received request. At least for now, it would be sufficient to update only the models that already have a fromJson method. These are the models we are likely to want to store and debug. I’m not sure if we need these methods for all models.

Of course, the team will make the most appropriate decision.
